src/expr/src/scalar/func.rs

```diff
@@ -1841,23 +1841,21 @@ where
     }
```

Comment:
meta: for this commit, could you add a brief explanation of what the optimization is and why it's legal?